snappy_pipeline/workflows/somatic_neoepitope_prediction/__init__.py

```diff
@@ -31,11 +31,338 @@
+    sheet_shortcut_class = CancerCaseSheet
+    sheet_shortcut_kwargs = {
+        "options": CancerCaseSheetOptions(allow_missing_normal=True, allow_missing_tumor=True)
+    }
+
+    @classmethod
+    def default_config_yaml(cls):
+        """Return default config YAML, to be overwritten by project-specific one."""
+        return DEFAULT_CONFIG
+
+    def __init__(self, workflow, config, config_lookup_paths, config_paths, workdir):
+        super().__init__(
+            workflow,
+            config,
+            config_lookup_paths,
+            config_paths,
+            workdir,
+            (
+                SomaticVariantCallingWorkflow,
+                SomaticVariantAnnotationWorkflow,
+                NgsMappingWorkflow,
+            ),
+        )
+        # Register sub step classes so the sub steps are available
+        self.register_sub_step_classes((NeoepitopePreparationStepPart, LinkOutStepPart))
+        self.register_sub_workflow(
+            "somatic_variant_annotation",
+            self.w_config["step_config"]["somatic_neoepitope_prediction"][
+                "path_somatic_variant_annotation"
+            ],
+        )
+        self.register_sub_workflow(
+            "ngs_mapping",
+            self.w_config["step_config"]["somatic_neoepitope_prediction"]["path_rna_ngs_mapping"],
+        )
+        # Copy over "tools" setting from somatic_variant_calling/ngs_mapping if not set here
+        if not self.w_config["step_config"]["somatic_neoepitope_prediction"]["tools_ngs_mapping"]:
+            self.w_config["step_config"]["somatic_neoepitope_prediction"][
+                "tools_ngs_mapping"
+            ] = self.w_config["step_config"]["ngs_mapping"]["tools"]["dna"]
+        if not self.w_config["step_config"]["somatic_neoepitope_prediction"][
+            "tools_somatic_variant_calling"
+        ]:
+            self.w_config["step_config"]["somatic_neoepitope_prediction"][
+                "tools_somatic_variant_calling"
+            ] = self.w_config["step_config"]["somatic_variant_calling"]["tools"]
+        if not self.w_config["step_config"]["somatic_neoepitope_prediction"][
+            "tools_somatic_variant_annotation"
+        ]:
+            self.w_config["step_config"]["somatic_neoepitope_prediction"][
+                "tools_somatic_variant_annotation"
+            ] = ["vep"]
+        if not self.w_config["step_config"]["somatic_neoepitope_prediction"]["tools_rna_mapping"]:
+            self.w_config["step_config"]["somatic_neoepitope_prediction"][
+                "tools_rna_mapping"
+            ] = self.w_config["step_config"]["ngs_mapping"]["tools"]["rna"]
+        if not self.w_config["step_config"]["somatic_neoepitope_prediction"]["preparation"][
+            "path_features"
+        ]:
+            self.w_config["step_config"]["somatic_neoepitope_prediction"]["preparation"][
+                "path_features"
+            ] = self.w_config["static_data_config"]["features"]["path"]
```

Comment:
`self.w_config["step_config"]["somatic_neoepitope_prediction"]` is the same as `self.config` (at least content-wise). Replacing that should make this a lot shorter and easier to read.